Adds a fuzzing test

LeCodeurDuDimanche opened the pull request:
The test check multiple function calls of the public API in a random order and with random inputs over a large set of randomly generated valid Starname objects
This PR also adds some additional tests to validate the `Filter` method

LeCodeurDuDimanche committed:
Add unit tests to Filter

LeCodeurDuDimanche committed:
Add a fuzz test

davepuchyr commented on internal/test/fuzz_test.go:
```diff
@@ -0,0 +1,222 @@
+	crud "github.com/iov-one/cosmos-sdk-crud"
+)
+
+const domainMinLength = 4
```
Please use the starname constraint parameters from below and add `uri` and `resource` tests.
```
      "valid_domain_name": "^[-a-z0-9_]{4,32}$",
      "valid_account_name": "^[-.a-z0-9_]{1,63}$",
      "valid_uri": "^[-a-z0-9]{1,15}:[-a-z0-9]{1,40}$",
      "valid_resource": "^[-.a-z0-9A-Z/:_ #]{1,200}$",
```
(Those parameters are from our mainnet; `curl -X POST "https://lcd-private-iov-mainnet-2.iov.one/configuration/query/configuration"`)

LeCodeurDuDimanche replied:
Do you want me to use those parameters as they are defined now or to fetch them from the mainnet in the test ?
I don't know how I have to test uri and resource, as these attributes does not seem to be secondary keys of the Starname object (if so they are not used with `cosmos-sdk-crud`)

davepuchyr replied:
Just use the hard-coded parameters for now.

You're right about the `uri`, it's not indexed; however, `resource` is on the mainnet.  The Starname object isn't a perfect representation of the data on the mainnet.

LeCodeurDuDimanche replied:
So I add a `resource` field to the `TestStarname` object in order to test it ?

davepuchyr replied:
Yes, that's probably the path of least resistance.

LeCodeurDuDimanche committed:
Add a resource field and use the up-to-date validation regex for generation

Adds a `resource` field to the TestStarname object and integrate it into the fuzzing test. Regenerated the type definitions to reflect the modified protobuf definition.
The starnames are now generated directly from the current validation regexes of the mainnet, which are fetched from https://lcd-private-iov-mainnet-2.iov.one/configuration/query/configuration

LeCodeurDuDimanche committed:
Make the code compliant with go version under 1.16
